flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -494,12 +486,12 @@ public void requestPartitions() throws IOException, InterruptedException {
 	// ------------------------------------------------------------------------
 
 	@Override
-	public Optional<BufferOrEvent> getNextBufferOrEvent() throws IOException, InterruptedException {
+	public Optional<BufferOrEvent> getNext() throws IOException, InterruptedException {
 		return getNextBufferOrEvent(true);
```

Comment:
I notice here still have a choice for blocking or non-blocking, I check the code path, does this only used for the batch task?(just a question)

Reply:
I think yes. Blocking method looks like it is only used in `AbstractRecordReader` and/or in tests.